openfisca_france/model/prelevements_obligatoires/impot_revenu/ir.py

```diff
@@ -2712,18 +2786,19 @@ def function(self, simulation, period):
         veuf = simulation.calculate('veuf', period)
         celibataire_ou_divorce = simulation.calculate('celibataire_ou_divorce', period)
         nbptr = simulation.calculate('nbptr', period)
-        ppe = simulation.legislation_at(period.start).ir.credits_impot.ppe
+        ppe = simulation.legislation_at(period.start).impot_revenu.credits_impot.ppe
 
         seuil = (veuf | celibataire_ou_divorce) * (ppe.eligi1 + 2 * max_(nbptr - 1, 0) * ppe.eligi3) \
                 + maries_ou_pacses * (ppe.eligi2 + 2 * max_(nbptr - 2, 0) * ppe.eligi3)
         return period, (rfr * ppe_coef) <= seuil
 
 
-class ppe_rev(Variable):
+class ppe_rev(DatedVariable):
     column = FloatCol
     entity_class = Individus
     label = u"ppe_rev"
 
+    @dated_function(stop = date(2015, 12, 31))
```

Comment:
There are benefits to have **one** canonical way to do one thing: unified codebase, better learning curve, and simpler change. As said in [The zen of python](https://www.python.org/dev/peps/pep-0020/): 
> There should be one-- and preferably only one --obvious way to do it.

So, for a specific recurring need, we should agree on a preferred implementing way.

Here the need is to compute a variable only before a certain date. The most obvious way to do it seems to just provide the date the formula should stop at. Using `DatedVariable` brings unnecessary complexity.

About #547, I don't think `DatedVariables` deprecation is on the scope of the PR. But I'm open to discussions about building a simpler way to handle formula changes over time ;).